Fix pp calculate error when circle count is 0

cdwcgt opened the pull request:
this is one of the beatmap link that will cause this problem:
https://osu.ppy.sh/beatmapsets/1713744#osu/3549141

this pr will let `RhythmComplexity` skill return 0 when circle count is 0

It is worth noting that I don't know why `normalisedHitError` will be NaN if count 300 is less then circle count, it will cause all skill to 0, so I adjust it to 0 but not NaN

cdwcgt committed:
do not calculate accuracy if no object need accuracy

cdwcgt committed:
why NaN here?

Syriiin commented on osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs:
```diff
@@ -75,7 +75,7 @@ private static double calculateNormalisedHitError(double od, int objectCount, in
             int circle300Count = count300 - (objectCount - circleCount);
             if (circle300Count <= 0)
-                return double.NaN;
+                return 0;
```
Changing this has an impact on output.
https://github.com/Syriiin/osu/blob/9f1f39f52548c43a425f581e19920ef07b9306e4/osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs#L122
With this change, it would result in `560` instead of `0`.
It could be refactored, but I don't think that's relevant to this bug, so let's revert this for now.

Syriiin replied:
Reverted in fa30ebb83afa8fce31c447c878e54cba23610cb6

cdwcgt replied:
https://osu.ppy.sh/beatmapsets/1713744#osu/3544157
check this map, except SS or this will return double.NaN to `normalisedHitError` and all skill will be 0 pp

![image](https://github.com/Syriiin/osu/assets/34775378/b3a2a606-eee4-42e3-bf22-32d0333783d4)

Syriiin commented on osu.Game.Rulesets.Osu/Difficulty/Skills/RhythmComplexity.cs:
```diff
@@ -14,7 +14,7 @@ namespace osu.Game.Rulesets.Osu.Difficulty.Skills
     public class RhythmComplexity : Skill
     {
-        private int circleCount;
+        private int amountHitObjectsWithAccuracy;
```
This is a good name change, however I'm going to suggest we revert it for now, because now that lazer has accuracy for sliders too, this isn't accurate anymore.
If we want to adjust the code to include sliders if scorev2 is active, then this would be a good change.
Let's revert that that for now as `circleCount` is actually accurate, even if not semantically named.

Syriiin replied:
Reverted in 8f073b65f380476858a1de6401fcea6a0c55eee5

Syriiin committed:
Revert change from NaN to 0

Syriiin committed:
Revert var name change

Syriiin commented:
> It is worth noting that I don't know why normalisedHitError will be NaN if count 300 is less then circle count, it will cause all skill to 0, so I adjust it to 0 but not NaN

This is a good point actually. I don't think setting it to 0 would actually make sense, since that would imply millisecond perfect timing in this case, instead of what actually happens here, which is worst case scenario timing.
Setting this to the hit window of a 50 is probably the easiest fix here to make these scores not return 0, but not be incorrectly inflated either.

Syriiin committed:
Return 50 hit window instead of NaN if no circles

If we can't guarentee any 300s on circles, the easiest fallback is the worst possible scenario of a 50 hit window. Having this as NaN meant in this case, all skills would be zeroed out because they were weighted by the normalised hit error.

Syriiin committed:
Fix 50 hit window values
